### PR TITLE
Support for multiple s3 buckets

### PR DESCRIPTION
**Pull request by suhrawardi**
Please have a look at it. Feedback is very welcome!

### CODE REVIEW

**Comment by jesuisbonbon** [comment]
+1

**Comment by xjunior** [comment]
+1

**Comment by katbow** [comment]
+1, looks good.

@stavro @suhrawardi any updates on this one?

File: lib/arc/storage/s3.ex [hunk 1]
```diff
@@ -38,8 +39,9 @@ defmodule Arc.Storage.S3 do
   defp ensure_keyword_list(map) when is_map(map), do: Map.to_list(map)
 
   # If the file is stored as a binary in-memory, send to AWS in a single request
-  defp do_put(file=%Arc.File{binary: file_binary}, s3_key, s3_options) when is_binary(file_binary) do
-    ExAws.S3.put_object(bucket(), s3_key, file_binary, s3_options)
+  defp do_put(file=%Arc.File{binary: file_binary}, s3_config) when is_binary(file_binary) do
```
Comment (pedep) [inline lib/arc/storage/s3.ex]: You can move the destructuring in the function parameters instead
```elixir
def do_put(file, {s3_bucket, s3_key, s3_options}) do
```

File: lib/arc/storage/s3.ex [hunk 2]
```diff
@@ -48,12 +50,13 @@ defmodule Arc.Storage.S3 do
   end
 
   # Stream the file and upload to AWS as a multi-part upload
-  defp do_put(file, s3_key, s3_options) do
+  defp do_put(file, s3_config) do
+    {s3_bucket, s3_key, s3_options} = s3_config
```
Comment (pedep) [inline lib/arc/storage/s3.ex]: Same as above

**Comment by suhrawardi** [comment]
Thx for your feedback @pedep, I updated the PR!

**Pull request merged**